numpy/ma/core.py

```diff
@@ -3837,7 +3837,7 @@ def fill_value(self, value=None):
                 DeprecationWarning, stacklevel=2)
 
         _fill_value = self._fill_value
-        if _fill_value is None:
+        if _fill_value is None or np.isscalar(target):
             # Create the attribute if it was undefined
             self._fill_value = target
```

Comment:
In general `np.isscalar` is a dubious function unfortunately. We want to make sure that `self._fill_value` is always an `ndarray`. With this, it might end up being a scalar (the opposite check may make sense: If it is already an array, use copyto/`self._fill_value[...] = `).

(I am not sure if that works always even for scalars, it may at least with unsafe casting.  But of course we have that integer fill-value cast problem, which may actually end up in this code path?)

Reply:
Oh, I apologize for misunderstanding your point. It appears that the issue is related to `ndarray`, which is beyond my current abilities to resolve. I will close the PR later (the PR couldn't fix the bug) and will add this information to the original issue for others to see. Thank you for taking the time to review this.